src/app/shared/messages-area/messages-area.component.ts

```diff
@@ -21,7 +22,14 @@ export class MessagesAreaComponent implements AfterViewInit {
 	ngAfterViewInit():void {
 		this.messagesAreaService.addMessageEmitter.subscribe(
 			( message ):void => {
-				this.messages.push( message );
+				this.messages.forEach( ( messageAux )=> {
+					if( messageAux.statusCode !== message.statusCode ) return;
```

Comment:
What happens if there are two messages with same statusCode but different messages?
Should we decide to compare also other Message properties like title, content, etc? (Just a question)
Also, a global boolean (`messageExists`) to ask if it already exists doesn't seem necessary, it's just being used within this scope.
Also, check the [`array.some`](https://developer.mozilla.org/es/docs/Web/JavaScript/Referencia/Objetos_globales/Array/some) operator to ask for a boolean matching a condition within an array.